sound/hda/ext/hdac_ext_stream.c

```diff
@@ -310,13 +311,41 @@ hdac_ext_host_stream_assign(struct hdac_bus *bus,
 		spin_lock_irq(&bus->reg_lock);
 		res->hstream.opened = 1;
 		res->hstream.running = 0;
-		res->hstream.substream = substream;
+		res->hstream.substream = NULL;
+		res->hstream.cstream = NULL;
 		spin_unlock_irq(&bus->reg_lock);
 	}
 
 	return res;
 }
 
+static struct hdac_ext_stream *
+hdac_ext_host_stream_pcm_assign(struct hdac_bus *bus,
+				struct snd_pcm_substream *substream)
+{
+	struct hdac_ext_stream *res;
+
+	res = hdac_ext_host_stream_assign(bus, substream->stream);
+	if (res)
+		res->hstream.substream = substream;
+
+	return res;
+}
+
+struct hdac_ext_stream *
+hdac_ext_host_stream_compr_assign(struct hdac_bus *bus,
+				struct snd_compr_stream *cstream)
+{
+	struct hdac_ext_stream *res;
+
+	res = hdac_ext_host_stream_assign(bus, cstream->direction);
+	if (res)
+		res->hstream.cstream = cstream;
+
+	return res;
+}
+EXPORT_SYMBOL_GPL(hdac_ext_host_stream_compr_assign);
```

Comment:
shouldn't probe be supported even in cases where we don't use the hdac library?

we should try and make this dual license.

Reply:
As stated in PR's body:

> Another point of contention are probe/ compress operations themselves. As these are basically made of hdac_foos and IPCs only, I decided to make this feature Intel specific for now.

Unsure how to do without HDA. Basically, all probe compr operations are assigned via snd_soc_cdai_ops to snd_soc_dai_driver::ops of given dai driver instance.
DAI declaration is are system specific (e.g. skl_dai) and so my understanding is: DAIs ops are too.

In order to make probes truly generic, I'd need more data - again, without HDA, there is nothing left of probes.

Reply:
of course we need the HDAudio DMAs. The point was about using the HDAC library, which is GPLv2. So to meet the dual license requirements, the SOF code redeveloped all the DMA access. We only use the hdac library when HDMI or HDAudio codecs support is required.